src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -181,7 +181,7 @@ internal static InferredLambdaReturnType InferReturnType(ArrayBuilder<(RefKind,
                     // to the caller, and up through MethodTypeInferrer.Infer.
                     bestResultType = hadNullabilityMismatch ?
                         default :
-                        TypeSymbolWithAnnotations.Create(bestType, isNullableIfReferenceType: BestTypeInferrer.GetIsNullable(resultTypes));
+                        TypeSymbolWithAnnotations.Create(bestType, isNullableIfReferenceType: BestTypeInferrer.GetIsNullable(resultTypes), fromDeclaration: true);
```

Comment:
>fromDeclaration: true [](start = 139, length = 21)

The calculation of lambda return type is re-done during flow analysis, so it feels like we shouldn't always use "fromDeclaration: true".

Reply:
> The calculation of lambda return type is re-done during flow analysis, so it feels like we shouldn't always use "fromDeclaration: true".

Some of the changes are simply preserving existing behavior, often relying on existing tests to verify that. The current plan is to remove all APIs that use "bool?" to create TypeSymbolWithAnnotations. Annotations from flow analysis will naturally flow through here after the change. 

---
In reply to: [230530045](https://github.com/dotnet/roslyn/pull/30913#discussion_r230530045) [](ancestors = 230530045)